### ISSUE TITLE
Config option not working

### ISSUE THREAD

**Issue by joezimjs** [danielhusar/gulp-stylint#13]
When I pass nothing or an empty object to the plugin as an argument (e.g. `stylint()` or `stylint({})`) linting happens using the base rules, which generates some warnings for me. This is expected. I prefer not to keep my .xxxrc files in the root directory, so I move them somewhere else, so I try to use the `config` option to point to my rules file, but when I do that, it seems to disable all rules and gives no warnings to me. I even tried with no `config` option and used a `strict:true` option and I got the same behavior. If I understand correctly, using `strict` should cause _more_ warnings. Using just `failOnError:true` seems to work though.

So yea, for whatever reason, I just can't seem to get my custom rules configuration to work.

**Comment by yoyurec** [comment]
+1

**Comment by SimenB** [comment]
Probably my fault somehow. The test passes, though... I'll see if I can fix it tonight!

**Comment by danielhusar** [comment]
Looks like we need to parse the config, and pass it as object to stylus for programatic usage. (also looks like strict mode was deprecated as I don't see it in stylus readme any more so we should get rid of it)

**Comment by danielhusar** [comment]
Should be fixed in `1.1.9`

**Comment by SimenB** [comment]
We should add a test as well. Cherry-pick the one @joezimjs made?

Ah, that was a PR that has been merged. great!

**Comment by yoyurec** [comment]
will you update readme or post here examples of usage?

```
        ...
        .pipe(stylint({
            options: {reporter: 'stylint-stylish'},
            failOnError: true
        }));
```

doesn't work

**Comment by SimenB** [comment]
Untested, but try

``` js
 .pipe(stylint({
            reporter: 'stylint-stylish',
            failOnError: true
        }));
```

**Comment by yoyurec** [comment]
nothing ))
just empty output (with core reporter all is ok - warnings shows)

**Comment by SimenB** [comment]
Ok, I'll take a look later tonight.

@danielhusar The fix here fails if there are comments in the config-file, btw

**Comment by yoyurec** [comment]
> fix here fails if there are comments in the config-file

comments? they are not allowed in json (if i understood your message correct)

**Comment by joezimjs** [comment]
@yoyurec 

> will you update readme or post here examples of usage?
> 
> ```
>    ...
>    .pipe(stylint({
>        options: {reporter: 'stylint-stylish'},
>        failOnError: true
>    }));
> ```
> 
> doesn't work

It's supposed to be `config`, not `options`.

**Comment by yoyurec** [comment]
i've tried all variations ))
and `config` do error: "TypeError: path must be a string"

**Comment by joezimjs** [comment]
That is a product of the fix that @danielhusar put in. He didn't check if the `config` option was an object or a string. He just assumed string and used it as the file path. I'll jump on that.

**Comment by SimenB** [comment]
> comments? they are not allowed in json (if i understood your message correct)

It's standard that rc-files handles it, though. Stylint itself ignores comments in latest version :smile: 

Awesome that you're on it @joezimjs!

**Comment by joezimjs** [comment]
@yoyurec try using 

```
   .pipe(stylint({
       config: {reporter: 'stylint-stylish'},
       failOnError: true
   }));
```

It should work now.

**Comment by SimenB** [comment]
@yoyurec `1.1.10` is the version you want to use :smile:

**Comment by yoyurec** [comment]
sorry, but still the same ((
is it works for you?

**Comment by danielhusar** [comment]
Sorry guys, I didnt noticed you can also pass object as config

**Comment by joezimjs** [comment]
@danielhusar Actually... I thought you could, and that's what @yoyurec was trying to do. The original implementation technically _only_ allowed you to pass it as an object. I added that back in for @yoyurec and added the possibility to the README. 

@yoyurec Did you `npm update`? What exactly do you have as your Gulp task code and what are you seeing for output?

...

Ok I just tested this by passing only the reporter information in to the `config` property and it didn't assume the default linting rules. This appears to be an issue related to Stylint, not the Gulp plugin, based on that, but I'm not 100% sure on that.

**Comment by joezimjs** [comment]
This is technically a separate issue, though, so we should probably create that.

**Comment by SimenB** [comment]
@yoyurec Could you create a separate issue?
And yes, this is probably an issue with stylint. I'll try to dig into it tonight. 

Edit: I think I figured it out. We pass options to the constructor, and those options short-circuit the config-setting. If we instead pass options to `init` (`create`-function), it might work better. I'll try it out later (at work now).

**Comment by yoyurec** [comment]
> @yoyurec Could you create a separate issue?

done! https://github.com/danielhusar/gulp-stylint/issues/17
